src/techui_builder/generate.py

```diff
@@ -157,6 +214,68 @@ def _create_widget(self, component: Entry) -> EmbeddedDisplay | ActionButton:
 
         return new_widget
 
+    def layout_widgets(self, widgets: list[EmbeddedDisplay | ActionButton]):
+        group_spacing = 30
+        max_group_height = 800
+        spacing_x = 20
+        spacing_y = 30
+        # Group tiles by size
+        groups = defaultdict(list)
```

Comment:
Is it possible to type-hint this? `sorted_groups` says is has "Unknown" types because there isn't any hinting currently, which causes all variables in the `for` loop to also have no linting.

Reply:
This seems to work.
```suggestion
        groups: dict[tuple[int, int], list[EmbeddedDisplay | ActionButton]] = (
            defaultdict(list)
        )
```

Reply:
Is there a reason why you are making `groups` into a dict, but then instantly converting it into a sorted list? (I can see why you used a `defaultdict`; so you can append to it easily)